compression/decompression-broken-chunks.html

```diff
@@ -0,0 +1,146 @@
+<!doctype html>
+<meta charset="utf-8">
+<div id=log></div>
+<script>
+
+ // This file confirms whether DecompressionStream returns a expected data
+ // or unexpected data if a compressed data is broken.
+
+ // Both deflated data and gzipped data has a header and a footer, and
+ // in them, there are some bytes which have their own roles, for example
+ // Compression Method(CM), FLG, and so on.
+ // If you want to know more details, see
+ // RFC1950 for deflate: https://tools.ietf.org/html/rfc1950
+ // RFC1952 for gzip: https://tools.ietf.org/html/rfc1952
+
+ // Expected decompressed string is 'expected output'.
+ const compressedArrayWithDeflate = [120, 156, 75, 173, 40, 72, 77, 46, 73, 77, 81, 200, 47, 45, 41, 40, 45, 1, 0, 48, 173, 6, 36];
+ const compressedArrayWithGzip = [31, 139, 8, 0, 0, 0, 0, 0, 0, 3, 75, 173, 40, 72, 77, 46, 73, 77, 81, 200, 47, 45, 41, 40, 45, 1, 0, 176, 1, 57, 179, 15, 0, 0, 0];
+
+ // The numbers in the following list represent roles of the byte offset like CM.
+ // for deflate
+ const deflateByteoffsetList = [
+   CMF = 0, FLG = 1, DICTID0 = 2, DICTID1 = 3, DICTID2 = 4,
```

Comment:
I think FLG.FDICT is not set, so these will not be DICTID bytes and instead are just the first 4 bytes of the DATA.

Reply:
Exactly! Thanks!